### PR TITLE
Validation intro and corrections of inline ValSem documentation

### LINKED ISSUE #568: Add more documentation about validation
Besides the documentation in the [wiki](https://github.com/openmls/openmls/wiki/Message-validation), more documentation could be added in the book and inline where the validation checks are performed.

### PR DESCRIPTION
**Pull request by kkohbrok**
Fixes #568.

### CODE REVIEW

File: book/src/message_validation.md [hunk 1]
```diff
@@ -1,14 +1,40 @@
...
 
-- Syntax validation: This should be mostly covered by the decoding
-- Semantic validation: Checks to make sure a message is valid in a given context (signature verification, epoch number check, etc.)
-- Group policy validation: checks about handshake type, etc.
-- AS/policy validation: Checks to see whether syntactically and semantically correct messages should be adopted or dropped (Is a member allowed to add another member? Is a member allowed to remove another member?)
+Validation is enforced using Rust's type system. The chain of functions used to process incoming messages is described in the chapter on [Processing incoming messages](user_manual/processing.md), where each function takes a distinct type as input and produces a distinct type as output, thus ensuring that the individual steps can't be skipped. We now detail which step performs which validation checks.
+
+### Syntax validation
+
+Incoming messages in the shape of a byte string can only be deserialized into a `MlsMessageIn` struct. Deserialization ensures that the message is a syntactically correct MLS message, i.e. either an MLSPlaintext or an MLSCiphertext.
```
Comment (raphaelrobert) [inline book/src/message_validation.md:11]:
```suggestion
Incoming messages in the shape of a byte string can only be deserialized into a `MlsMessageIn` struct. Deserialization ensures that the message is a syntactically correct MLS message, i.e. either an MLSPlaintext or an MLSCiphertext.
For the latter case, further syntax checks are applied once the message is decrypted.
```

File: book/src/message_validation.md [hunk 2]
```diff
@@ -1,14 +1,40 @@
...
+
+### Semantic validation
+
+Every function in the processing chain performs a number of semantic validation steps. For a list of these steps, see [below][message_validation.md#detailed-list-of-validation-steps]. In the following, we will give a brief overview over which function performs which category of checks.
```
Comment (raphaelrobert) [inline book/src/message_validation.md:15]:
```suggestion
Every function in the processing chain performs a number of semantic validation steps. For a list of these steps, see [below](message_validation.md#detailed-list-of-validation-steps). In the following, we will give a brief overview over which function performs which category of checks.
```

File: book/src/message_validation.md [hunk 3]
```diff
@@ -1,14 +1,40 @@
...
+* the message is properly authenticated by signature (`ValSem010`), membership tag (`ValSem008`) and confirmation tag (`ValSem205`),
+* proposals are valid relative to one-another and the current group state, e.g. no redundant adds or removes targeting non-members (`ValSem100`-`ValSem112`),
+* commits are valid relative to the group state and the proposals it covers (`ValSem200`-`ValSem205`) and
+* external commits are valid according to the spec (`ValSem240`-`ValSem245`, `ValSem247` is checked as part of `ValSem010`).
+
+After performing these steps, messages are returned as `ProcessedMessage`s that the application can either use immediately (application messages) or inspect and decide if they find them valid according to the application's own policy (proposals and commits). Proposals can then be stored in the proposal queue via `.store_pending_proposal()`, while commits can be merged into the group state via `.merge_staged_commit()`.
 
 ## Detailed list of validation steps
 
+The following is a list of the individual semantic validation steps performed by OpenMLS.
```
Comment (raphaelrobert) [inline book/src/message_validation.md:36]:
```suggestion
The following is a list of the individual semantic validation steps performed by OpenMLS, including the location of the tests.
```

**Review by raphaelrobert — APPROVED** [review]
Thanks for the thorough review! Just some suggestions

**Review by kkohbrok — COMMENTED** [review]
Excellent suggestions, thanks!

**Pull request merged**